iceoryx_posh/test/moduletests/test_popo_event_variable.cpp

```diff
@@ -0,0 +1,259 @@
+
+using namespace ::testing;
+using namespace iox::popo;
+using namespace iox::units::duration_literals;
+
+class EventVariable_test : public Test
+{
+  public:
+    using Type_t = iox::cxx::BestFittingType_t<iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET>;
+    using NotificationVector_t = iox::cxx::vector<Type_t, iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET>;
+
+    const iox::ProcessName_t m_process{"Ferdinand"};
+    EventVariableData m_eventVarData{m_process};
+
+    const iox::units::Duration m_timeToWait = 2_s;
+};
+
+TEST_F(EventVariable_test, AllNotificationsAreFalseAfterConstruction)
+{
+    EventVariableData sut;
+    for (auto& notification : sut.m_activeNotifications)
+    {
+        EXPECT_THAT(notification, Eq(false));
+    }
+}
+
+TEST_F(EventVariable_test, CorrectProcessNameAfterConstructionWithProcessName)
+{
+    EXPECT_THAT(m_eventVarData.m_process.c_str(), StrEq(m_process));
+}
+
+TEST_F(EventVariable_test, AllNotificationsAreFalseAfterConstructionWithProcessName)
+{
+    for (auto& notification : m_eventVarData.m_activeNotifications)
+    {
+        EXPECT_THAT(notification, Eq(false));
+    }
+}
+
+TEST_F(EventVariable_test, NotifyActivatesCorrectIndex)
+{
+    constexpr Type_t EVENT_INDEX = iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET - 1U;
+    EventNotifier sut(m_eventVarData, EVENT_INDEX);
+    sut.notify();
+    for (Type_t i = 0U; i < iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET; i++)
+    {
+        if (i == EVENT_INDEX)
+        {
+            EXPECT_THAT(m_eventVarData.m_activeNotifications[i], Eq(true));
+        }
+        else
+        {
+            EXPECT_THAT(m_eventVarData.m_activeNotifications[i], Eq(false));
+        }
+    }
+}
+
+TEST_F(EventVariable_test, NotifyActivatesNoIndexIfIndexIsTooLarge)
+{
+    constexpr Type_t EVENT_INDEX = iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET;
+    EventNotifier sut(m_eventVarData, EVENT_INDEX);
+    sut.notify();
+    for (const auto& notification : m_eventVarData.m_activeNotifications)
+    {
+        EXPECT_THAT(notification, Eq(false));
+    }
+}
+
+TEST_F(EventVariable_test, WaitIsNonBlockingAfterDestroyAndReturnsEmptyVector)
+{
+    EventListener sut(m_eventVarData);
+    sut.destroy();
+    const auto& activeNotifications = sut.wait();
+
+    EXPECT_THAT(activeNotifications.size(), Eq(0U));
+}
+
+TEST_F(EventVariable_test, WaitIsNonBlockingAfterDestroyAndNotifyAndReturnsEmptyVector)
+{
+    EventListener sut(m_eventVarData);
+    sut.destroy();
+
+    EventNotifier notifier(m_eventVarData, 0U);
+    notifier.notify();
+
+    const auto& activeNotifications = sut.wait();
+    EXPECT_THAT(activeNotifications.size(), Eq(0U));
+}
+
+TEST_F(EventVariable_test, DestroyWakesUpWaitWhichReturnsEmptyVector)
+{
+    EventListener sut(m_eventVarData);
+
+    NotificationVector_t activeNotifications;
+
+    std::thread waiter([&] {
+        activeNotifications = sut.wait();
+        EXPECT_THAT(activeNotifications.size(), Eq(0U));
+    });
+
+    sut.destroy();
```

Comment:
this is a potential deadlock. Please use the sepukku class with `std::terminate` to prevent this.